layouts/BlogLayout.js

```diff
@@ -0,0 +1,96 @@
+import Link from '@/components/Link'
+import Tag from '@/components/Tag'
+import siteMetadata from '@/data/siteMetadata'
+import { useState } from 'react'
+import Pagination from '@/components/Pagination'
+
+const postDateTemplate = { year: 'numeric', month: 'long', day: 'numeric' }
+
+export default function BlogLayout({ posts, title, pagination, prevDisable, nextDisable }) {
+  const [searchValue, setSearchValue] = useState('')
+  const filteredBlogPosts = posts.filter((frontMatter) => {
```

Comment:
This breaks the search implementation. Ideally, it should work across all the pages. Search results should also be paginated if they exceed a certain length

Reply:
Right now, the search function searches for a keyword in the current page only.
Is this how you want?
Or, you want the search bar to find keyword and display results from content in `/blog/2` or `/blog/3`?

Reply:
That's because in the prev implementation all the blogs get displayed in a single page. But now that it is split across multiple pages the search for blogs should still cover all blog posts. So the later behaviour would be good. But it should still search only a single page for the tags (because you already filtered on a category)